File: expire.py
import datetime
import os

import boto3
import dateutil

MAX_AGE = int(os.environ["MAX_AGE"])
PARAMS = {
    "Filters": [{"Name": "volume-id",
                 "Values": os.environ["VOLUMES"].split(" ")}]
}


def max_age():
    now = datetime.datetime.utcnow().replace(tzinfo=dateutil.tz.tzutc())
    return now - datetime.timedelta(days=MAX_AGE)


def lambda_handler(event, context):
    ec2 = boto3.client('ec2', region_name=os.environ["AWS_REGION"])
    shots = ec2.describe_snapshots(**PARAMS)
    old = max_age()

    count = 0
    for snap in shots["Snapshots"]:
        if snap["StartTime"] < old:
            count += 1
            ec2.delete_snapshot(SnapshotId=snap['SnapshotId'])

    print("Deleted {} snapshot(s)".format(count))

File: expire.js
var AWS = require('aws-sdk');

var MAX_AGE = parseInt(process.env.MAX_AGE);
var PARAMS = {
  Filters: [{Name: "volume-id",
             Values: process.env.VOLUMES.split(" ")}]
};

function daysOld(time) {
  return (new Date() - time)/(1000*60*60*24);
}

exports.handler = (event, context, callback) => {
  var ec2 = new AWS.EC2({region: process.env.AWS_REGION});

  ec2.describeSnapshots(PARAMS).promise()
    .then(data => Promise.all(
        data.Snapshots
          .filter(snap => daysOld(snap.StartTime) > MAX_AGE)
          .map(snap => ec2.deleteSnapshot({SnapshotId: snap.SnapshotId}).promise())
    ))
    .then(results => {
      console.log(`Deleted ${results.length} snapshot(s)`)
      callback(null, `Deleted ${results.length} snapshot(s)`);
    })
    .catch(err => {
      console.log(err, err.stack);
      callback(err);
    });
};
